google-cloud-core/src/main/java/com/google/cloud/Policy.java

```diff
@@ -221,7 +220,12 @@ public final Builder setBindings(List<Binding> bindings) {
       return this;
     }
 
-    /** Removes the role (and all identities associated with that role) from the policy. */
+    /**
+     * Removes the role (and all identities associated with that role) from the policy.
+     *
+     * @throws IllegalArgumentException if policy version is equal to 3 or has conditional bindings
+     *     because conditional policies are not supported
+     */
     public final Builder removeRole(Role role) {
       checkArgument(
```

Comment:
I still think all of these should use `checkState` instead of `checkArgument` so they throw `IllegalStateException`.

Throw `IllegalArgumentException` when the caller should have known that the arguments to this method are invalid.
Throw `IllegalStateException` when the caller should have known that this method cannot be called given the state of the object.